javamelody-core/src/main/resources/net/bull/javamelody/resource/monitoring.js

```diff
@@ -0,0 +1,237 @@
+	  });
+	
+	  $$('.selectDatabaseReport').invoke("observe", "change", function(){
+	  	window.location.href='?part=database&request=' + this.selectedIndex;
+	  });
+	
+	  $$('.selectDeploymentPeriod').invoke("observe", "change", function(){
+	  	document.deploymentPeriodForm.submit();
+	  });
+
+	  $$("[data-margin-left-px]").each(function(element){
+	  	element.style.marginLeft=element.getAttribute("data-margin-left-px") + "px";
+	  });
+
+	  $$(".rumData td[data-width-percent]").each(function(element){
+	  	element.style.width=element.getAttribute("data-width-percent") + "%";
+	  });
+
+	  if(document.customPeriodForm) {
+		var test = document.createElement('input'); test.type = 'date';
+		if(test.type === 'text') {
+		  document.customPeriodForm.pattern.value = '';
+		  document.getElementById('customPeriodPattern').style.display='inline';
+		}
+		function validateCustomPeriodForm() {
+		   periodForm = document.customPeriodForm;
+		   if (periodForm.startDate.value.length == 0) {
+		      alert('Dates are mandatory');
+		      periodForm.startDate.focus();
+		      return false;
+		   }
+		   if (periodForm.endDate.value.length == 0) {
+		      alert('Dates are mandatory');
+		      periodForm.endDate.focus();
+		      return false;
+		   }
+		   periodForm.period.value=periodForm.startDate.value + '|' + periodForm.endDate.value;
+		   return true;
+		}
+		$$('form[name="customPeriodForm"]').invoke("observe", "submit", function(event){
+			if(! validateCustomPeriodForm()){
+			  event.preventDefault();
+		    }
+		});
+	  }
+	
+	  $$('a.back').invoke("observe", "click", function(event){
+	    history.back();
+	  	event.preventDefault();
+	  });
+	  
+	  $$('.copyHash').invoke("observe", "click", function(event){
+	    document.getElementById('hash').select();
+	    document.getElementById('hash').setSelectionRange(0, 99999); /*For mobile devices*/
+	    document.execCommand('copy');
+	  	event.preventDefault();
+	  });
+
+	  $$('.confirm').invoke("observe", "click", function(event){
+	  	if(! confirm(this.getAttribute('data-confirm'))){
+	  		event.preventDefault();
+  		}
+	  });
+
+	  $$('.showHide').invoke("observe", "click", function(event){
+	  	var id = this.getAttribute("data-show-hide-id");
+	  	if(!id) id = this.id;
+	  	showHide(id);
+	  	event.preventDefault();
+	  });
+
+	  $$('.customPeriod').invoke("observe", "click", function(event){
+		showHide('customPeriod');
+		document.customPeriodForm.startDate.focus();
+	  	event.preventDefault();
+	  });
+
+	  $$('.addApplication').invoke("observe", "click", function(event){
+		showHide('addApplication');
+		document.appForm.appName.focus();
+	  	event.preventDefault();
+	  });
+
+	  $$('.addAggregation').invoke("observe", "click", function(event){
+		showHide('addAggregation');
+		document.aggregationForm.appName.focus();
+	  	event.preventDefault();
+	  });
+
+	  $$('.deploymentPeriod').invoke("observe", "click", function(event){
+		showHide('customPeriod');
```

Comment:
if this was used, it would have displayed the custom period form instead of the deployment period form